pydantic_ai_slim/pydantic_ai/models/outlines.py

```diff
@@ -0,0 +1,561 @@
+
+        return filtered_settings
+
+    def _format_llama_cpp_inference_kwargs(self, model_settings: dict[str, Any]) -> dict[str, Any]:
+        """Select the model settings supported by the LlamaCpp model."""
+        supported_args = [
+            'max_tokens',
+            'temperature',
+            'top_p',
+            'seed',
+            'presence_penalty',
+            'frequency_penalty',
+            'logit_bias',
+            'extra_body',
+        ]
+        filtered_settings = {k: model_settings[k] for k in supported_args if k in model_settings}
+
+        return filtered_settings
+
+    def _format_mlxlm_inference_kwargs(  # pragma: no cover
+        self, model_settings: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Select the model settings supported by the MLXLM model."""
+        supported_args = [
+            'extra_body',
+        ]
+        filtered_settings = {k: model_settings[k] for k in supported_args if k in model_settings}
+
+        return filtered_settings
+
+    def _format_sglang_inference_kwargs(self, model_settings: dict[str, Any]) -> dict[str, Any]:
+        """Select the model settings supported by the SGLang model."""
+        supported_args = [
+            'max_tokens',
+            'temperature',
+            'top_p',
+            'presence_penalty',
+            'frequency_penalty',
+            'extra_body',
+        ]
+        filtered_settings = {k: model_settings[k] for k in supported_args if k in model_settings}
+
+        return filtered_settings
+
+    def _format_vllm_offline_inference_kwargs(  # pragma: no cover
+        self, model_settings: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Select the model settings supported by the vLLMOffline model."""
+        from vllm.sampling_params import SamplingParams  # pyright: ignore
+
+        supported_args = [
+            'max_tokens',
+            'temperature',
+            'top_p',
+            'seed',
+            'presence_penalty',
+            'frequency_penalty',
+            'logit_bias',
+            'extra_body',
+        ]
+        # The arguments that are part of the fields of `ModelSettings` must be put in a `SamplingParams` object and
+        # provided through the `sampling_params` argument to vLLM
+        sampling_params = model_settings.get('extra_body', {}).pop('sampling_params', SamplingParams())
+
+        for key in supported_args:
+            setattr(sampling_params, key, model_settings.get(key, None))
+
+        filtered_settings = {
+            'sampling_params': sampling_params,
+            **model_settings.get('extra_body', {}),
+        }
+
+        return filtered_settings
+
+    async def _format_prompt(self, messages: list[ModelMessage], instructions: str | None) -> Chat:  # noqa: C901
+        """Turn the model messages into an Outlines Chat instance."""
+        chat = Chat()
+
+        if instructions:
```

Comment:
This reminds me we should support `ModelResponse.instructions` as well. You can use `Model._get_instructions`, there are examples of its usage in other model classes.

Reply:
My bad, I meant `ModelRequest`! But yes that's the logic we need to make sure we don't lose https://ai.pydantic.dev/agents/#instructions